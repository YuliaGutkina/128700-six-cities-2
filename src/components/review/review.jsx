import React from 'react';
import moment from "moment";
import {Rating} from "../rating/rating";
import {ComplexPropType} from "../../types/types";


export const Review = (props) => {
  const {commentData} = props;

  return <li className="reviews__item">
    <div className="reviews__user user">
      <div className="reviews__avatar-wrapper user__avatar-wrapper">
        <img className="reviews__avatar user__avatar" src={commentData.user.avatar} width={54} height={54} alt="Reviews avatar" />
      </div>
      <span className="reviews__user-name">
        {commentData.user.name}
      </span>
    </div>
    <div className="reviews__info">
      <Rating
        className="reviews__rating"
        starsClassName="reviews__stars"
        value={commentData.rating}
      />
      <p className="reviews__text">
        {commentData.comment}
      </p>
      <time className="reviews__time" dateTime={commentData.date}>{moment(commentData.date).format(`MMMM YYYY`)}</time>
    </div>
  </li>;
};

Review.propTypes = {
  commentData: ComplexPropType.COMMENT
};
