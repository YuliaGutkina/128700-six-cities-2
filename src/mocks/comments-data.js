export const commentsData = [
  {
    id: 1,
    user: {
      id: 4,
      isPro: false,
      name: `Max`,
      avatar: `img/1.png`
    },
    rating: 4,
    comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    date: `2019-05-08T14:13:56.569Z`
  },
  {
    id: 2,
    user: {
      id: 4,
      isPro: false,
      name: `Max`,
      avatar: `img/1.png`
    },
    rating: 4,
    comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    date: `2019-05-08T14:13:56.569Z`
  },
  {
    id: 3,
    user: {
      id: 4,
      isPro: false,
      name: `Max`,
      avatar: `img/1.png`
    },
    rating: 4,
    comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    date: `2019-05-08T14:13:56.569Z`
  }
];
