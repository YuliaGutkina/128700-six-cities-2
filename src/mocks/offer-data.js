export const offersData = [
  {
    id: 1,
    city: {
      name: `Amsterdam`,
      location: {
        latitude: 52.370216,
        longitude: 4.895168,
        zoom: 10
      }
    },
    preview: `img/1.png`,
    images: [`img/1.png`, `img/2.png`],
    title: `Beautiful & luxurious studio at great location`,
    isFavorite: false,
    isPremium: false,
    rating: 4.8,
    type: `apartment`,
    bedrooms: 3,
    maxAdults: 4,
    price: 120,
    goods: [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
    host: {
      id: 3,
      isPro: true,
      name: `Angelina`,
      avatar: `img/1.png`
    },
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    location: {
      latitude: 52.35514938496378,
      longitude: 4.673877537499948,
      zoom: 8
    }
  },
  {
    id: 2,
    city: {
      name: `Amsterdam`,
      location: {
        latitude: 52.370216,
        longitude: 4.895168,
        zoom: 10
      }
    },
    preview: `img/1.png`,
    images: [`img/1.png`, `img/2.png`],
    title: `Beautiful & luxurious studio at great location`,
    isFavorite: false,
    isPremium: false,
    rating: 4.8,
    type: `apartment`,
    bedrooms: 3,
    maxAdults: 4,
    price: 120,
    goods: [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
    host: {
      id: 3,
      isPro: true,
      name: `Angelina`,
      avatar: `img/1.png`
    },
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    location: {
      latitude: 52.35514938496378,
      longitude: 4.673877537499948,
      zoom: 8
    }
  },
  {
    id: 3,
    city: {
      name: `Tel Aviv`,
      location: {
        latitude: 52.370216,
        longitude: 4.895168,
        zoom: 10
      }
    },
    preview: `img/1.png`,
    images: [`img/1.png`, `img/2.png`],
    title: `Beautiful & luxurious studio at great location`,
    isFavorite: false,
    isPremium: false,
    rating: 4.8,
    type: `apartment`,
    bedrooms: 3,
    maxAdults: 4,
    price: 120,
    goods: [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
    host: {
      id: 3,
      isPro: true,
      name: `Angelina`,
      avatar: `img/1.png`
    },
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    location: {
      latitude: 52.35514938496378,
      longitude: 4.673877537499948,
      zoom: 8
    }
  },
  {
    id: 4,
    city: {
      name: `Amsterdam`,
      location: {
        latitude: 52.370216,
        longitude: 4.895168,
        zoom: 10
      }
    },
    preview: `img/1.png`,
    images: [`img/1.png`, `img/2.png`],
    title: `Beautiful & luxurious studio at great location`,
    isFavorite: false,
    isPremium: false,
    rating: 4.8,
    type: `apartment`,
    bedrooms: 3,
    maxAdults: 4,
    price: 120,
    goods: [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
    host: {
      id: 3,
      isPro: true,
      name: `Angelina`,
      avatar: `img/1.png`
    },
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    location: {
      latitude: 52.35514938496378,
      longitude: 4.673877537499948,
      zoom: 8
    }
  },
  {
    id: 5,
    city: {
      name: `Amsterdam`,
      location: {
        latitude: 52.370216,
        longitude: 4.895168,
        zoom: 10
      }
    },
    preview: `img/1.png`,
    images: [`img/1.png`, `img/2.png`],
    title: `Beautiful & luxurious studio at great location`,
    isFavorite: false,
    isPremium: false,
    rating: 4.8,
    type: `apartment`,
    bedrooms: 3,
    maxAdults: 4,
    price: 120,
    goods: [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
    host: {
      id: 3,
      isPro: true,
      name: `Angelina`,
      avatar: `img/1.png`
    },
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    location: {
      latitude: 52.35514938496378,
      longitude: 4.673877537499948,
      zoom: 8
    }
  },
  {
    id: 6,
    city: {
      name: `Amsterdam`,
      location: {
        latitude: 52.370216,
        longitude: 4.895168,
        zoom: 10
      }
    },
    preview: `img/1.png`,
    images: [`img/1.png`, `img/2.png`],
    title: `Beautiful & luxurious studio at great location`,
    isFavorite: false,
    isPremium: false,
    rating: 4.8,
    type: `apartment`,
    bedrooms: 3,
    maxAdults: 4,
    price: 120,
    goods: [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
    host: {
      id: 3,
      isPro: true,
      name: `Angelina`,
      avatar: `img/1.png`
    },
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    location: {
      latitude: 52.35514938496378,
      longitude: 4.673877537499948,
      zoom: 8
    }
  },
  {
    id: 7,
    city: {
      name: `Amsterdam`,
      location: {
        latitude: 52.370216,
        longitude: 4.895168,
        zoom: 10
      }
    },
    preview: `img/1.png`,
    images: [`img/1.png`, `img/2.png`],
    title: `Beautiful & luxurious studio at great location`,
    isFavorite: false,
    isPremium: false,
    rating: 4.8,
    type: `apartment`,
    bedrooms: 3,
    maxAdults: 4,
    price: 120,
    goods: [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
    host: {
      id: 3,
      isPro: true,
      name: `Angelina`,
      avatar: `img/1.png`
    },
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    location: {
      latitude: 52.35514938496378,
      longitude: 4.673877537499948,
      zoom: 8
    }
  },
  {
    id: 8,
    city: {
      name: `Paris`,
      location: {
        latitude: 52.370216,
        longitude: 4.895168,
        zoom: 10
      }
    },
    preview: `img/1.png`,
    images: [`img/1.png`, `img/2.png`],
    title: `Beautiful & luxurious studio at great location`,
    isFavorite: false,
    isPremium: false,
    rating: 4.8,
    type: `apartment`,
    bedrooms: 3,
    maxAdults: 4,
    price: 120,
    goods: [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
    host: {
      id: 3,
      isPro: true,
      name: `Angelina`,
      avatar: `img/1.png`
    },
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    location: {
      latitude: 52.35514938496378,
      longitude: 4.673877537499948,
      zoom: 8
    }
  },
  {
    id: 9,
    city: {
      name: `Paris`,
      location: {
        latitude: 52.370216,
        longitude: 4.895168,
        zoom: 10
      }
    },
    preview: `img/1.png`,
    images: [`img/1.png`, `img/2.png`],
    title: `Beautiful & luxurious studio at great location`,
    isFavorite: false,
    isPremium: false,
    rating: 4.8,
    type: `apartment`,
    bedrooms: 3,
    maxAdults: 4,
    price: 120,
    goods: [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
    host: {
      id: 3,
      isPro: true,
      name: `Angelina`,
      avatar: `img/1.png`
    },
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    location: {
      latitude: 52.35514938496378,
      longitude: 4.673877537499948,
      zoom: 8
    }
  }
];
