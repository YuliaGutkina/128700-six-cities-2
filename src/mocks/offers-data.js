export const offersData = [
  {
    city: `Amsterdam`,
    initialCoordinates: [52.38333, 4.9],
    places: [
      {
        title: `Beautiful & luxurious apartment at great location`,
        type: `Apartment`,
        isPremium: true,
        src: `img/apartment-01.jpg`,
        price: 120,
        rating: 93,
        coordinates: [52.3909553943508, 4.85309666406198]
      },
      {
        title: `Wood and stone place`,
        type: `Private room`,
        isPremium: false,
        src: `img/room.jpg`,
        price: 80,
        rating: 80,
        coordinates: [52.369553943508, 4.85309666406198]
      },
      {
        title: `Canal View Prinsengracht`,
        type: `Apartment`,
        isPremium: false,
        src: `img/apartment-02.jpg`,
        price: 132,
        rating: 80,
        coordinates: [52.3909553943508, 4.929309666406198]
      },
      {
        title: `Nice, cozy, warm big bed apartment`,
        type: `Apartment`,
        isPremium: true,
        src: `img/apartment-03.jpg`,
        price: 180,
        rating: 100,
        coordinates: [52.3809553943508, 4.939309666406198]
      }
    ]
  },
  {
    city: `Paris`,
    initialCoordinates: [48.864716, 2.349014],
    places: []
  },
  {
    city: `Hamburg`,
    initialCoordinates: [53.551086, 9.993682],
    places: []
  }
];
