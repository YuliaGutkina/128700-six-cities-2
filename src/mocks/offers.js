export const offers = [
  {
    title: `Beautiful & luxurious apartment at great location`,
    type: `Apartment`,
    isPremium: true,
    src: `img/apartment-01.jpg`,
    price: 120,
    rating: 93
  },
  {
    title: `Wood and stone place`,
    type: `Private room`,
    isPremium: false,
    src: `img/room.jpg`,
    price: 80,
    rating: 80
  },
  {
    title: `Canal View Prinsengracht`,
    type: `Apartment`,
    isPremium: false,
    src: `img/apartment-02.jpg`,
    price: 132,
    rating: 80
  },
  {
    title: `Nice, cozy, warm big bed apartment`,
    type: `Apartment`,
    isPremium: true,
    src: `img/apartment-03.jpg`,
    price: 180,
    rating: 100
  }
];
