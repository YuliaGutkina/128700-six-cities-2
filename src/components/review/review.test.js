import React from "react";
import ShallowRenderer from 'react-test-renderer/shallow';

import {Review} from "./review";
import {commentsData} from "../../mocks/comments-data";


it(`Bookmark correctly renders after relaunch`, () => {
  const renderer = new ShallowRenderer();
  renderer.render(<Review
    commentData={commentsData[0]}
  />);
  const result = renderer.getRenderOutput();

  expect(result).toMatchSnapshot();
});
