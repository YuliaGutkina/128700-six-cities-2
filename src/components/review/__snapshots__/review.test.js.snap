// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bookmark correctly renders after relaunch 1`] = `
<li
  className="reviews__item"
>
  <div
    className="reviews__user user"
  >
    <div
      className="reviews__avatar-wrapper user__avatar-wrapper"
    >
      <img
        alt="Reviews avatar"
        className="reviews__avatar user__avatar"
        height={54}
        src="img/1.png"
        width={54}
      />
    </div>
    <span
      className="reviews__user-name"
    >
      Max
    </span>
  </div>
  <div
    className="reviews__info"
  >
    <Rating
      className="reviews__rating"
      starsClassName="reviews__stars"
      value={4}
    />
    <p
      className="reviews__text"
    >
      A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
    </p>
    <time
      className="reviews__time"
      dateTime="2019-05-08T17:13:56+03:00"
    >
      May 2019
    </time>
  </div>
</li>
`;
